crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheMetadata.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.cache;
+
+import com.google.cloud.tools.crepecake.image.DuplicateLayerException;
+import com.google.cloud.tools.crepecake.image.ImageLayers;
+import com.google.cloud.tools.crepecake.image.LayerPropertyNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Hold information for what data the cache holds. */
```

Comment:
This comment is a little weirdly worded?

Also, how is CacheMetadata going to be used? Is this some reflection of the filesystem (and the fs is the source of truth?) or is it going to be edited to written back down?

Reply:
The cache will store all the cached blobs as well as a `metadata.json` file that has information describing each blob in the cache. The `CacheMetadata` class will be loaded from `metadata.json` and then changed over the course of the build and written back to the fs.